crates/ty_python_semantic/resources/mdtest/attributes.md

```diff
@@ -985,6 +985,33 @@ class Foo:
         y = x
 ```
 
+When accessing an attribute in a staticmethod, we don't suggest the use of `self.`.
+
+```py
+class Foo:
+    def __init__(self):
+        self.x = 42
+
+    @staticmethod
+    def static_method():
+        # error: [unresolved-reference] "Name `x` used when not defined"
+        y = x
+```
+
+When accessing an attribute in a classmethod, we suggest the use of `cls.`.
```

Comment:
I think the other thing to test in a classmethod is that we don't suggest anything at all if the variable name matches an instance-only attribute. E.g. if we add an `__init__` method below that has `self.y = 1` and then reference `y` in the classmethod, we should not suggest `cls.y` (or anything at all) for `y`